mahjong/tile/definition.py

```diff
@@ -116,3 +115,43 @@ def tile_triplet(tile: Tile) -> TileSet:
 
 def tile_flush(tile: Tile) -> TileSet:
     return tile.flush()
+
+
+class AkaTile(Tile):
+    _ATILE_POOL = {}
+
+    def __new__(cls, number: int, color: str):
+        tile_hash = tile_hash_num(color, number)
+        if tile_hash in AkaTile._ATILE_POOL:
+            return AkaTile._ATILE_POOL[tile_hash]
+        else:
+            tile_hash = tile_hash_num(color, number + 5)
```

Comment:
Do we need to reject any creation of non-5 aka tiles, to ensure no one will happened to create a **Aka 6s** by passing (1s) to this constructor? I mean, shall we `assert number == 0` at first?